lib/internal/modules/esm/worker.js

```diff
@@ -164,24 +182,45 @@ async function customizedModuleWorker(lock, syncCommPort, errorHandler) {
     setImmediate(() => {});
   });
 
+  const allThreadRegisteredHandlerPorts = [];
+  /**
+   * @callback registerHandler
+   * @param {MessagePort} toWorkerThread - Upon Worker creation a message channel between the new Worker
+   * and the Hooks thread is being initialized. This is the MessagePort that the Hooks thread will use post
+   * messages to the worker. The other MessagePort is passed to the new Worker itself via LOAD_SCRIPT message
+   */
+  function registerHandler(toWorkerThread) {
```

Comment:
Is there no unregister needed if a worker ends?
My feeling is that `allThreadRegisteredHandlerPorts` just grows.

Reply:
good point, pushed a fix for this with last commit.